Build error on Perlmutter

@jeff-cohere I'm getting a similar error on Crusher, but the notes below are for reproducing the error on Perlmutter. I believe @jedbrown was encountering a similar error.

1. Load relevant modules on Perlmutter and set the path to PETSc
```
module purge
module load craype-x86-milan
module load libfabric/1.15.2.0
module load craype-network-ofi
module load xpmem/2.5.2-2.4_3.48__gd0f7936.shasta
module load cpe/22.11
module load xalt/2.10.2
module load gpu/1.0
module load evp-patch
module load python/3.9-anaconda-2021.11
module load cray-dsmml/0.2.2
module load PrgEnv-gnu/8.3.3
module load gcc/11.2.0
module load Nsight-Compute/2022.1.1
module load Nsight-Systems/2022.2.1
module load cudatoolkit/11.5
module load craype-accel-nvidia80
module load cray-libsci/22.11.1.2
module load craype/2.7.19
module load cray-mpich/8.1.22
module load cray-hdf5-parallel/1.12.2.1
module load cray-netcdf-hdf5parallel/4.9.0.1
module load cray-parallel-netcdf/1.12.3.1
module load cmake/3.22.0

export PETSC_DIR=/global/cfs/projectdirs/m4267/petsc/petsc_99e66fd4349
export PETSC_ARCH=arch-pm-gpu-debug-gcc-11-2-0-no-debug
export LD_LIBRARY_PATH=$PETSC_DIR/$PETSC_ARCH/lib:$LD_LIBRARY_PATH
```

2. Building RDycore
```
>git show -n 1
commit 6abbcf29282308741d36288c6e8f8b451f76c562 (HEAD -> main, origin/main, origin/HEAD)
Merge: 179e432 e351ac2
Author: Jeffrey N. Johnson <jeff@cohere-llc.com>
Date:   Wed Jun 7 12:46:46 2023 -0700

    Merge pull request #53 from RDycore/jeff-cohere/mat-props-2

    Added material properties to input parser and RDycore

mkdir $PETSC_ARCH

cd $PETSC_ARCH
cmake .. -DCMAKE_INSTALL_PREFIX=$PWD -DCMAKE_BUILD_TYPE=Debug
make -j4
Scanning dependencies of target rdycore_f90
[  2%] Creating directories for 'cmocka_proj'
[  4%] Creating directories for 'yaml_proj'
[  6%] Building Fortran object src/f90-mod/CMakeFiles/rdycore_f90.dir/rdycore.F90.o
[  8%] No download step for 'cmocka_proj'
[ 11%] No download step for 'yaml_proj'
[ 15%] No update step for 'yaml_proj'
[ 15%] No update step for 'cmocka_proj'
[ 17%] Linking Fortran static library librdycore_f90.a
[ 20%] No patch step for 'yaml_proj'
[ 22%] No patch step for 'cmocka_proj'
[ 24%] Performing configure step for 'yaml_proj'
[ 26%] Performing configure step for 'cmocka_proj'
[ 26%] Built target rdycore_f90
-- yaml_proj configure command succeeded.  See also /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/external/libyaml/src/yaml_proj-stamp/yaml_proj-configure-*.log
[ 28%] Performing build step for 'yaml_proj'
-- yaml_proj build command succeeded.  See also /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/external/libyaml/src/yaml_proj-stamp/yaml_proj-build-*.log
[ 31%] Performing install step for 'yaml_proj'
-- yaml_proj install command succeeded.  See also /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/external/libyaml/src/yaml_proj-stamp/yaml_proj-install-*.log
[ 33%] Completed 'yaml_proj'
[ 33%] Built target yaml_proj
[ 35%] Creating directories for 'cyaml_proj'
[ 37%] No download step for 'cyaml_proj'
[ 40%] No update step for 'cyaml_proj'
[ 42%] No patch step for 'cyaml_proj'
[ 44%] Performing configure step for 'cyaml_proj'
[ 46%] Performing build step for 'cyaml_proj'
CMake Error at src/cyaml_proj-stamp/cyaml_proj-build-Debug.cmake:37 (message):
  Command failed: 2

   'env' 'PREFIX=/global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug' 'CFLAGS=-I/global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/include' 'LDFLAGS=-L/global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/lib -lyaml' 'make' 'all' 'VARIANT=debug'

  See also

    /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/external/libcyaml/src/cyaml_proj-stamp/cyaml_proj-build-*.log


-- stdout output is:

-- stderr output is:
make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
/usr/bin/ld: /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/lib/libyaml.a(api.c.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:157: build/debug/libcyaml.so.2] Error 1

CMake Error at src/cyaml_proj-stamp/cyaml_proj-build-Debug.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [external/CMakeFiles/cyaml_proj.dir/build.make:86: external/libcyaml/src/cyaml_proj-stamp/cyaml_proj-build] Error 1
make[1]: *** [CMakeFiles/Makefile2:1038: external/CMakeFiles/cyaml_proj.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
-- cmocka_proj configure command succeeded.  See also /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/external/cmocka/src/cmocka_proj-stamp/cmocka_proj-configure-*.log
[ 48%] Performing build step for 'cmocka_proj'
-- cmocka_proj build command succeeded.  See also /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/external/cmocka/src/cmocka_proj-stamp/cmocka_proj-build-*.log
[ 51%] Performing install step for 'cmocka_proj'
-- cmocka_proj install command succeeded.  See also /global/cfs/projectdirs/m4267/gbisht/rdycore-ceed/build-arch-pm-gpu-debug-gcc-11-2-0-no-debug/external/cmocka/src/cmocka_proj-stamp/cmocka_proj-install-*.log
[ 53%] Completed 'cmocka_proj'
[ 53%] Built target cmocka_proj
make: *** [Makefile:146: all] Error 2
```

--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -7,6 +7,8 @@ include(ExternalProject)
 # we need this for some finicky library installation paths
 include(GNUInstallDirs)
 
+find_package(Patch REQUIRED)
+
 #----------------------------------------
 # Build libyaml, a YAML parsing library.
 #----------------------------------------
@@ -51,6 +53,12 @@ ExternalProject_Add(cyaml_proj
                     LOG_INSTALL TRUE
                     LOG_OUTPUT_ON_FAILURE TRUE
                     BUILD_BYPRODUCTS ${PROJECT_BINARY_DIR}/lib/${LIBCYAML_LIB_NAME})
+ExternalProject_Add_Step(cyaml_proj patch_makefile
+                         COMMAND ${Patch_EXECUTABLE} Makefile ${CMAKE_CURRENT_SOURCE_DIR}/libcyaml.Makefile.patch
+                         COMMENT "Patching libcyaml/Makefile to bypass shared objects..."
+                         DEPENDEES configure
+			 DEPENDERS build
+                         WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/libcyaml)
 add_library(cyaml SHARED IMPORTED GLOBAL)
 set_target_properties(cyaml PROPERTIES
   IMPORTED_LOCATION ${PROJECT_BINARY_DIR}/lib/${LIBCYAML_LIB_NAME})

--- a/external/libcyaml.Makefile.patch
+++ b/external/libcyaml.Makefile.patch
@@ -1,0 +1,26 @@
+diff --git a/Makefile b/Makefile
+index 613cf56..bad0d02 100644
+--- a/Makefile
++++ b/Makefile
+@@ -126,7 +126,7 @@ TEST_BINS = \
+ 		$(BUILDDIR)/test/units/cyaml-shared \
+ 		$(BUILDDIR)/test/units/cyaml-static
+ 
+-all: $(BUILDDIR)/$(LIB_SH_MAJ) $(BUILDDIR)/$(LIB_STATIC) examples
++all: $(BUILDDIR)/$(LIB_STATIC) examples
+ 
+ coverage: test-verbose
+ 	$(Q)$(MKDIR) $(BUILDDIR)
+@@ -173,11 +173,8 @@ docs:
+ clean:
+ 	rm -rf build/
+ 
+-install: $(BUILDDIR)/$(LIB_SH_MAJ) $(BUILDDIR)/$(LIB_STATIC) $(BUILDDIR)/$(LIB_PKGCON)
++install: $(BUILDDIR)/$(LIB_STATIC) $(BUILDDIR)/$(LIB_PKGCON)
+ 	$(INSTALL) -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
+-	$(INSTALL) $(BUILDDIR)/$(LIB_SH_MAJ) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(LIB_SH_VER)
+-	(cd $(DESTDIR)$(PREFIX)/$(LIBDIR) && { ln -s -f $(LIB_SH_VER) $(LIB_SH_MAJ) || { rm -f $(LIB_SH_MAJ) && ln -s $(LIB_SH_VER) $(LIB_SH_MAJ); }; })
+-	(cd $(DESTDIR)$(PREFIX)/$(LIBDIR) && { ln -s -f $(LIB_SH_VER) $(LIB_SHARED) || { rm -f $(LIB_SHARED) && ln -s $(LIB_SH_VER) $(LIB_SHARED); }; })
+ 	$(INSTALL) $(BUILDDIR)/$(LIB_STATIC) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(LIB_STATIC)
+ 	chmod 644 $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(LIB_STATIC)
+ 	$(INSTALL) -d $(DESTDIR)$(PREFIX)/$(INCLUDEDIR)/cyaml

